src/ISBootloaderAPP.cpp

```diff
@@ -48,14 +48,24 @@ eImageSignature cISBootloaderAPP::check_is_compatible()
     is_comm_instance_t comm;
     uint8_t buffer[2048];
     is_comm_init(&comm, buffer, sizeof(buffer));
-    int messageSize;
+    int messageSize, n, i, m;
+
+    // clear the Rx serial port
+    n = is_comm_free(&comm);
+    for (i = 0; i < 20; i++) 
```

Comment:
This seems a little non-deterministic, and unclear (but I follow its intent).  It maybe clearer to write this as a while() loop, with a retry-count that breaks to prevent locking.  Also, I generally recommend using an actual timeout, with a timer, rather than an arbitrary retry counter.

Reply:
Commenting section to clarify code